buildenv/jenkins/JenkinsfileBase

```diff
@@ -1154,13 +1154,15 @@ def uploadToArtifactory(pattern, artifactoryUploadDir="") {
 			artifactoryRepo = repos[0].trim()
 		}
 
-		def artifactoryRoorDir = params.ARTIFACTORY_ROOT_DIR ? params.ARTIFACTORY_ROOT_DIR : getJenkinsDomain()
-		if (artifactoryRoorDir.endsWith("/")) {
-			artifactoryRoorDir = artifactoryRoorDir.substring(0, artifactoryRoorDir.length() - 1);
+		def artifactoryRootDirBackup = "${artifactoryRepo}/".contains(getJenkinsDomain()) ? "" : getJenkinsDomain()
```

Comment:
Can we just simplify the logic to

```
def artifactoryRoorDir = ""
if (params.ARTIFACTORY_ROOT_DIR) {
   artifactoryRoorDir = params.ARTIFACTORY_ROOT_DIR
} else {
   // only set root dir to Jenkins domain if artifactoryRepo does not contain Jenkins domain root dir
   if (!artifactoryRepo.contains("/" + getJenkinsDomain())) {
     artifactoryRootDir = getJenkinsDomain()
   }
}
```

Reply:
Updated the logic and the new test link is [here](https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Grinder_Artifactory/12/console)